third_party/tlx/language/barrier.py

```diff
@@ -18,15 +18,44 @@ def alloc_barriers(
     """
     return tlx.mbarriers(_builder.create_alloc_barriers(num_barriers.value, arrive_count.value), )
 
+
 @tl.builtin
 def barrier_expect_bytes(
     bar: tlx.mbarriers,
     size: tl.constexpr,
     _builder=None,
 ) -> None:
     """
-    Signal a barrier of an expected number of bytes to be copied 
+    Signal a barrier of an expected number of bytes to be copied
     """
 
     # TODO. add validator logics
     _builder.create_barrier_expect(bar.handle, size.value)
+
+
+@tl.builtin
+def barrier_wait(
+    bar: tlx.buffered_tensor,
+    phase: tl.constexpr,
+    _builder=None,
+) -> None:
+    """
+    Signal a barrier of an expected number of bytes to be copied
+    """
+
+    # TODO. add validator logics
+    _builder.create_barrier_wait(bar.handle, phase.value)
+
+
+@tl.builtin
+def barrier_arrive(
+    bar: tlx.buffered_tensor,
+    arrive_count: tl.constexpr,
```

Comment:
A value of one for `arrive_count` is very common. Consider making the parameter optional with default value of one?